environment/src/main/java/com/sequenceiq/environment/environment/flow/config/update/event/EnvStackConfigUpdatesFailedEvent.java

```diff
@@ -0,0 +1,20 @@
+package com.sequenceiq.environment.environment.flow.config.update.event;
+
+import static com.sequenceiq.environment.environment.flow.config.update.event.EnvStackConfigUpdatesStateSelectors.FAILED_ENV_STACK_CONIFG_UPDATES_EVENT;
+
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.flow.reactor.api.event.BaseFailedFlowEvent;
+
+public class EnvStackConfigUpdatesFailedEvent extends BaseFailedFlowEvent implements Selectable {
```

Comment:
Yeah.  Removed

Reply:
Now, this class is referenced but it is never instantiated. So the flow will never trigger a failure action because this type is the only type that is selected by `FAILED_ENV_STACK_CONIFG_UPDATES_EVENT`. This means that `STACK_CONFIG_UPDATES_FAILED_STATE` is logically unreachable.

I think the action code or handler code should have a `catch (Exception e)` which creates an `EnvStackConfigUpdatesFailedEvent` and sends that event.

Reply:
Yeah. made some assumptions we talked about.  Added